contracts/CBOR.sol

```diff
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: MIT
 
-pragma solidity 0.6.12;
+pragma solidity 0.8.4;
 pragma experimental ABIEncoderV2;
```

Comment:
No need to include this pragma on ^0.8.0, as ABIEncoderV2 is activated by default. If you want to be specific, `pragma abicoder v2;` should be used nistead.

Reply:
According to Slither, the following methods should be declared as external (gas optimization):

- `decodeFixed16Array(CBOR.Value)`
- `decodeInt128Array(CBOR.Value)`
- `CBOR.decodeStringArray(CBOR.Value)`
- `CBOR.decodeUint64Array(CBOR.Value)`
- `CBOR.valueFromBytes(bytes)`

Reply:
In function `valueFromBuffer(BufferLib.Buffer)`, variable `length` is neither initialized nor ever assigned a value. Therefore, length field of returned CBOR.Value will always be 0.

Reply:
@aesedepece :middle_finger:

Reply:
After discussing, remove line #247 and modify #272 with a `0` :)